### PR TITLE
fix: llmguidance support tiktoken

### PR DESCRIPTION
**Pull request by JamesBrianD**
## Motivation

Fix SGL-59 https://github.com/sgl-project/sglang-jax/issues/424

## Accuracy Tests

```
sky exec sgl-jax-tpu-v6e-32 -d -- "cd sglang-jax && source .venv/bin/activate  && JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache uv run python -u -m sgl_jax.launch_server --model-path /models/xai-grok-2 --trust-remote-code --tp-size=32 --device=tpu --random-seed=3 --mem-fraction-static=0.8 --chunked-prefill-size=8192 --download-dir=/dev/shm --dtype=bfloat16 --max-running-requests=128 --skip-server-warmup --page-size=128 --tokenizer-path /models/xai-grok-2/tokenizer.tok.json  --dist-init-addr=10.128.0.94:10015 --nnodes=8 --node-rank=\${SKYPILOT_NODE_RANK}"
```

image

## Checklist

- [x] Please use English, otherwise it will be closed.
- [x] The purpose of the PR, or link existing issues this PR will resolve.
- [ ] The test plan, such as providing test command.
- [ ] (Optional) The necessary documentation update.

### CODE REVIEW

**Comment by JamesBrianD** [comment]
@jimoosciuc help to review this
